src/main/java/com/openshift/internal/restclient/model/Lifecycle.java

```diff
@@ -0,0 +1,65 @@
+package com.openshift.internal.restclient.model;
+
+import com.openshift.restclient.model.IHandler;
+import com.openshift.restclient.model.ILifecycle;
+import org.jboss.dmr.ModelNode;
+
+import java.util.Collections;
+import java.util.Optional;
+
+import static com.openshift.internal.util.JBossDmrExtentions.get;
+
+/**
+ * @author Ulf Lilleengen
+ */
+public class Lifecycle implements ILifecycle {
```

Comment:
I would consider extending ModelNodeAdapter which is like a light-weight KubernetesResource that allows you to contain the ModelNode.  This will allow you to remove fromJson and simply clone or provide an alternative Constructor that takes a ModelNode